dojo/tools/wiz/parser.py

```diff
@@ -1,91 +1,209 @@
+        }
+
+        for row in reader:
+            # Common fields
+            vulnerability_id = row.get("Name")
+            package_name = row.get("DetailedName")
+            package_version = row.get("Version")
+            severity = row.get("VendorSeverity")
+            finding_id = row.get("ID")
+
+            description = self._construct_string_field(description_fields, row)
+            mitigation = self._construct_string_field(mitigation_fields, row)
+            status_dict = self._convert_status(row)
+            # Create the finding object
+            finding = Finding(
+                title=f"{package_name}: {vulnerability_id}",
+                description=description,
+                mitigation=mitigation,
+                severity=self._validate_severities(severity),
+                static_finding=True,
+                unique_id_from_tool=finding_id,
+                component_name=package_name,
+                component_version=package_version,
+                date=date_parser.parse(row.get("FirstDetected")),
+                test=test,
+                **status_dict,
+            )
+            finding.unsaved_vulnerability_ids = [vulnerability_id]
+            finding.unsaved_tags = self._parse_tags(row.get("Tags", "[]"))
+            findings.append(finding)
+        return findings
+
+    def _construct_string_field(self, fields: dict[str, str], row: dict) -> str:
+        """Construct a formatted string based on the fields dict supplied."""
+        return_string = ""
+        for field, pretty_field in fields.items():
+            if (field_value := row.get(field)) is not None and len(field_value) > 0:
+                return_string += f"**{pretty_field}**: `{field_value}`\n"
+        return return_string
+
+    def _parse_tags(self, tags: str) -> list[str]:
+        """parse the Tag string dict, and convert to a list of strings.
+
+        The format of the tags is is "{""key"":""value""}" format
+        """
+        # Convert the string to a dict
+        tag_dict = json.loads(tags)
+        return [f"{key}: {value}" for key, value in tag_dict.items()]
+
+    def _validate_severities(self, severity: str) -> str:
+        """Ensure the supplied severity fits what DefectDojo is expecting."""
+        if severity not in SEVERITIES:
+            logger.error(f"Severity is not supported: {severity}")
+            # Default to Info severity
+            return "Info"
+        return severity
+
+    def _convert_status(self, row: dict) -> dict:
+        """Convert the "FindingStatus" column to a dict of Finding statuses.
+
+        - Open-> Active = True
+        - Other statuses that may exist...
+        """
+        if (status := row.get("FindingStatus")) is not None:
+            if status == "Open":
+                return {"active": True}
+        # Return the default status of active
+        return {"active": True}
+
+
+class WizParser(
+    WizParserByTitle,
+    WizParserByDetailedName,
+):
```

Comment:
It's not hurting anything as-is, but we probably don't need inheritance here?

```suggestion
class WizParser:
```

Reply:
Yeah, not strictly necessary. Will accept the suggestions. Some of the tests are stuck anyways